python/pyarrow/types.py

```diff
@@ -41,6 +41,7 @@
                    _INTERVAL_TYPES)
 _UNION_TYPES = {lib.Type_SPARSE_UNION, lib.Type_DENSE_UNION}
 _NESTED_TYPES = {lib.Type_LIST, lib.Type_FIXED_SIZE_LIST, lib.Type_LARGE_LIST,
+                 lib.Type_LIST_VIEW, lib.Type_LARGE_LIST_VIEW,
                  lib.Type_STRUCT, lib.Type_MAP} | _UNION_TYPES
```

Comment:
There is a  `is_nested` function in C++, perhaps we can use it instead of relying on hand-maintained constants on the Python side? Same for some of the other predicates actually (I see that `_INTERVAL_TYPES` is only holding a single kind of interval?)

Reply:
Good point. I filed https://github.com/apache/arrow/issues/40282